test/unit/app_logger_test.rb

```diff
@@ -0,0 +1,61 @@
+require 'test_helpers'
+
+class AppLoggerTest < UnitTest
+
+  def test_filtered_log_level_array_size
+    assert_raises ArgumentError do |exception|
+      AppLogger.filtered_log_level([:one_element])
+      assert_equal exception, 'Array must contain exactly 2 elements'
+    end
+  end
+
+  def test_filtered_log_level_error_element
+    assert_raises ArgumentError do |exception|
+      AppLogger.filtered_log_level([:weird_error, 'Error message'])
+      assert_equal exception, "First element must be one of the following: #{LOG_LEVELS.keys}"
+    end
+  end
+
+  def test_filtered_log_level_wrong_message
+    output, _ = capture_io do
+      AppLogger.filtered_log_level([:error, 'Test message']) do
+        begin
+          raise ArgumentError.new 'Test'
+        rescue Exception => e
+          mock_backtrace e, :error
+        end
+      end
+    end
+
+    assert_equal true, output.include?("[error] ArgumentError Test\n")
+  end
+
+  def test_filtered_log_level_wrong_severity
+    output, _ = capture_io do
+      AppLogger.filtered_log_level([:error, 'Test message']) do
+        begin
+          raise ArgumentError.new 'Test message'
+        rescue Exception => e
+          mock_backtrace e, :warn
+        end
+      end
+    end
+
+    assert_equal true, output.include?("[warn] ArgumentError Test message\n")
+  end
```

Comment:
I don't see a test where the one exception is suppressed and the other one(non-suppressed) is logged, which I think is the whole point of this PR

Reply:
This current setup is de facto MVP. Further development will be commited in following days.

Reply:
Test for new functionality should be present even before functionality is implemented. https://www.youtube.com/watch?v=ln4WnxX-wrw

Reply:
I know the basic of TDD. But Thank you.